modules/announcements/classes/announcement.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Elementor\Modules\Announcements\Classes;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Announcement {
+	/**
+	 * @var array
+	 */
+	protected $raw_data;
+	/**
+	 * @var array
+	 */
+	protected $triggers;
+
+	public function __construct( array $data ) {
+		$this->raw_data = $data;
+		$this->set_triggers();
+	}
+
+	/**
+	 * @return array
+	 */
+	protected function get_triggers(): array {
+		return $this->triggers;
+	}
+
+	protected function set_triggers() {
+		$triggers = $this->raw_data['triggers'] ?? [];
+		foreach ( $triggers as $trigger ) {
+			$this->triggers[] = Utils::get_trigger_object( $trigger );
+		}
+	}
+
+	/**
+	 * is_active
+	 * @return bool
+	 */
+	public function is_active(): bool {
+		$triggers = $this->get_triggers();
+
+		if ( empty( $triggers ) ) {
+			return true;
+		}
+
+		foreach ( $triggers as $trigger ) {
+			if ( ! $trigger->is_active() ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
```

Comment:
according to this logic a better name than "trigger" would be "condition" (or something similar) because you need all of them to be true to return true and if none exist then you also return true, but up to you

Reply:
It's more trigger than condition

Reply:
It's not CRITICAL, but I agree with Zeyad on this one.